CONTRIBUTING.rst

```diff
@@ -228,12 +228,21 @@ Guidelines
 * All changes are reviewed.  Ask on the
   `mailing list <http://groups.google.com/group/networkx-discuss>`_ if
   you get no response to your pull request.
+* Default dependencies are listed in ``requirements/default.txt`` and extra
+  (i.e., optional) dependencies are listed in ``requirements/extra.txt``.
+  We don't often add new default and extra dependencies.  Typically, new
+  proposed dependencies would first be added as extra dependencies.  
+  Extra dependencies should be easy to install on all platforms and widely-used.
+  New default dependencies should be easy to install on all platforms,
+  widely-used in the community, and have demonstrated potential for wide-spread
+  use in NetworkX.
```

Comment:
Maybe add that people who are considering adding code that has a dependency should consider adding a gallery example